src/polyline.js

```diff
@@ -54,7 +54,7 @@ polyline.decode = function(str, precision) {
         byte = null,
         latitude_change,
         longitude_change,
-        factor = Math.pow(10, precision || 5);
+        factor = Math.pow(10, typeof precision === 'number' ? precision : 5);
```

Comment:
Maybe we should use `Number.isInteger()` for this? Just because we don't want to use precisions that aren't whole numbers ?